packages/thirdweb/src/react/core/hooks/useTransactionDetails.ts

```diff
@@ -78,12 +82,33 @@ export function useTransactionDetails({
           encode(transaction).catch(() => "0x"),
         ]);
 
-      const [tokenInfo, gasCostWei] = await Promise.all([
+      const account = wallet?.getAccount();
+      if (!account) {
+        throw new Error("No active account");
+      }
+
+      const [tokenInfo, userBalance, gasCostWei] = await Promise.all([
         getToken(
           client,
           erc20Value ? erc20Value.tokenAddress : NATIVE_TOKEN_ADDRESS,
           transaction.chain.id,
         ).catch(() => null),
+        (async () =>
+          erc20Value &&
+          erc20Value.tokenAddress.toLowerCase() !== NATIVE_TOKEN_ADDRESS
+            ? getBalance({
+                contract: getContract({
+                  address: erc20Value.tokenAddress,
+                  chain: transaction.chain,
+                  client,
+                }),
+                address: account.address,
+              })
+            : getWalletBalance({
```

Comment:
get wallet Balance supports checking erc20, just need to pass the token address here